src/main/java/no/digipost/signature/client/core/internal/ClientExceptionMapper.java

```diff
@@ -30,6 +32,15 @@ public RuntimeException map(ProcessingException e) {
                         "Java 7 needs to be explicitly configured to support TLS 1.2. See 'JSSE tuning parameters' at https://blogs.oracle.com/java-platform-group/entry/diagnosing_tls_ssl_and_https.", e);
             }
         }
+
+        if (e.getCause() instanceof SSLHandshakeException) {
+            return new SignatureException("Unable to perform SSL handshake with remote server. Possible causes (could be others, see underlying error): \n" +
+                    "* Erroneous configuration of the trust store\n" +
+                    "* Intermediate network devices interfering with traffic (i.e. proxies)\n" +
+                    "* An attacker impersonating the server (man in the middle)\n" +
+                    "* Wrong TLS version. For Java 7, see 'JSSE tuning parameters' at https://blogs.oracle.com/java-platform-group/entry/diagnosing_tls_ssl_and_https for information about enabling the latest TLS versions", e);
```

Comment:
Maybe change the error message to include the sentence "Java 7 needs to be explicitly configured to support TLS 1.2." as in the case above, to be extra helpful i both cases?